tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc

```diff
@@ -80,116 +85,126 @@ class MklConvBwdFilterPrimitive : public MklPrimitive {
  public:
   explicit MklConvBwdFilterPrimitive(
       const MklConvBwdFilterParams& convBwdFilterDims)
-      : cpu_engine_(engine::cpu, 0) {
-    context_.bwd_filter_stream.reset(new stream(stream::kind::eager));
-    // create conv primitive
+      : cpu_engine_(ENGINE_CPU, 0) {
+    context_.bwd_filter_stream.reset(new CPU_STREAM(cpu_engine_));
+
+    // Create convolution backward filter primitive.
     if (context_.conv_bwd_filter == nullptr) {
       Setup(convBwdFilterDims);
     }
   }
 
   ~MklConvBwdFilterPrimitive() {}
 
-  // Convolution backward weights with bias
-  //   src_data:         input data buffer of src
-  //   diff_filter_data: output data buffer of diff_filter
-  //   diff_bias_data:   output data buffer of diff_bias
-  //   diff_dst_data:    input data buffer of diff_dst
+  // Convolution backward weights execution with bias
+  //   src_data:         input data buffer for src
+  //   diff_filter_data: output data buffer for diff_filter
+  //   diff_bias_data:   output data buffer for diff_bias
+  //   diff_dst_data:    input data buffer for diff_dst
   void Execute(const T* src_data, const T* diff_filter_data,
                const T* diff_bias_data, const T* diff_dst_data) {
     context_.src_mem->set_data_handle(
         static_cast<void*>(const_cast<T*>(src_data)));
     context_.diff_filter_mem->set_data_handle(
         static_cast<void*>(const_cast<T*>(diff_filter_data)));
-    context_.diff_bias_mem->set_data_handle(
-        static_cast<void*>(const_cast<T*>(diff_bias_data)));
+    if (diff_bias_data != nullptr) {
+      context_.diff_bias_mem->set_data_handle(
+          static_cast<void*>(const_cast<T*>(diff_bias_data)));
+    }
     context_.diff_dst_mem->set_data_handle(
         static_cast<void*>(const_cast<T*>(diff_dst_data)));
 
+#ifdef ENABLE_MKLDNN_V1
+    DCHECK_EQ(context_.bwd_filter_primitives.size(),
+              context_.bwd_filter_primitives_args.size());
+    for (size_t i = 0; i < context_.bwd_filter_primitives.size(); ++i) {
+      context_.bwd_filter_primitives.at(i).execute(
+          *context_.bwd_filter_stream,
+          context_.bwd_filter_primitives_args.at(i));
+    }
+#else
     context_.bwd_filter_stream->submit(context_.bwd_filter_primitives);
+#endif
```

Comment:
If this part gets refactored in https://github.com/tensorflow/tensorflow/pull/36496 ([comment](https://github.com/tensorflow/tensorflow/pull/36496/files/6e20ee6a00268a426f5006ab7cb3afc941d10895#r377373697)), please use the refactored method.

Reply:
In the other PR #36496, the comment says, refactoring can be done later since a bigger redesign is needed. I will also add a TODO here for refactoring when the design will be done.

Reply:
#36496 has been merged. Would you mind calling the refactored method (`execute_primitives`) in this PR? Thank you!